src/bedrock/hash_ring.cpp

```diff
@@ -165,25 +166,46 @@ vector<string> get_address_from_routing(
     unsigned& thread_id,
     unsigned& rid) {
 
+  int count = 0;
+
   communication::Key_Request key_req;
+  communication::Key_Response key_response;
   key_req.set_respond_address(ut.get_key_address_connect_addr());
   key_req.add_keys(key);
 
   string req_id = ip + ":" + to_string(thread_id) + "_" + to_string(rid);
   key_req.set_request_id(req_id);
-  rid += 1;
-
-  // query routing for addresses on the other tier
-  auto key_response = send_request<communication::Key_Request, communication::Key_Response>(key_req, sending_socket, receiving_socket, succeed);
   vector<string> result;
 
-  if (succeed) {
-    for (int j = 0; j < key_response.tuple(0).addresses_size(); j++) {
-      result.push_back(key_response.tuple(0).addresses(j));
+  int err_number = 1;
+
+  while (err_number == 1) {
+    if (count > 0 && count % 5 == 0) {
+      cerr << "Pausing for 5 seconds before continuing to query routing layer..." << endl;
+      usleep(5000000);
+    }
+
+    rid += 1;
+
+    // query routing for addresses on the other tier
+    key_response = send_request<communication::Key_Request, communication::Key_Response>(key_req, sending_socket, receiving_socket, succeed);
+
+    if (!succeed) {
+      return result;
+    } else {
+      cerr << "No servers have joined the cluster yet. Retrying request." << endl;
```

Comment:
why do we always print this error msg? Seems like if error number is 0 we are fine?

Reply:
Good call... fixing that. 😄